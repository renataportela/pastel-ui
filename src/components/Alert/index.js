export { default, ALERT_COLORS } from './Alert'