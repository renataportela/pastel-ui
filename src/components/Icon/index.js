export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';