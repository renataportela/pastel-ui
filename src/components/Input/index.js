export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
