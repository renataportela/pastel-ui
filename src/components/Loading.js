import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading