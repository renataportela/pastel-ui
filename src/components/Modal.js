import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;