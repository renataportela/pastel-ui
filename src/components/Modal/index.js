export { default } from './Modal'
export { default as ModalBase } from './ModalBase'