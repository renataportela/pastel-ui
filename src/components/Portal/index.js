export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'