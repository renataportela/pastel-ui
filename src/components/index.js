// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'