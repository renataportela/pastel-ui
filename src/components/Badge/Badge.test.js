import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, Badge, Button, Icon } from '~/components'

describe('Badge component', () => {
  it('renders text "badge" with badge "1562" and default color', () => {
    const tree = renderer
      .create(<Theme><Badge content="1562">badge</Badge></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders text "round" with round badge "12" and info color', () => {
    const tree = renderer
      .create(<Theme><Badge content="12" round color="info">round</Badge></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders text "icon" with Activity icon in a transparent badge', () => {
    const tree = renderer
      .create(
        <Theme>
          <Badge
            content={<Icon name="activity" size={16} color="red" />}
            bgColor="transparent"
          >
            icon
          </Badge>
        </Theme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a button with a dot badge with danger color', () => {
    const tree = renderer
      .create(
        <Theme>
          <Badge dot color="danger">
            <Button label="Button" />
          </Badge>
        </Theme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})