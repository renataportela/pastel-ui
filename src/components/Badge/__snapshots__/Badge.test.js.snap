// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component renders a button with a dot badge with danger color 1`] = `
<div
  className="Badge__Outer-dtpaqd-0 dJHSsV"
>
  <span
    className="BaseColors__BaseStyle-s1fbie-0 dtECaj Badge__BadgeIcon-dtpaqd-1 dBFPpa"
  >
    
  </span>
  <button
    className="ButtonStyle-sc-120zjl4-0 fJHZAc"
    color="primary"
    disabled={false}
    kind="default"
    size="md"
  >
    Button
    <span
      className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
      onMouseDown={[Function]}
    />
  </button>
</div>
`;

exports[`Badge component renders text "badge" with badge "1562" and default color 1`] = `
<div
  className="Badge__Outer-dtpaqd-0 dJHSsV"
>
  <span
    className="BaseColors__BaseStyle-s1fbie-0 cKbdWp Badge__BadgeIcon-dtpaqd-1 ikspmk"
  >
    
    1562
  </span>
  badge
</div>
`;

exports[`Badge component renders text "icon" with Activity icon in a transparent badge 1`] = `
<div
  className="Badge__Outer-dtpaqd-0 dJHSsV"
>
  <span
    className="BaseColors__BaseStyle-s1fbie-0 jqiDqM Badge__BadgeIcon-dtpaqd-1 ikspmk"
  >
    
    <svg
      className="Svg__SvgStyle-pyey0i-0 gYARyh"
      fill="none"
      height="24px"
      size={16}
      stroke="red"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="22 12 18 12 15 21 9 3 6 12 2 12"
      />
    </svg>
  </span>
  icon
</div>
`;

exports[`Badge component renders text "round" with round badge "12" and info color 1`] = `
<div
  className="Badge__Outer-dtpaqd-0 dJHSsV"
>
  <span
    className="BaseColors__BaseStyle-s1fbie-0 dZhYCv Badge__BadgeIcon-dtpaqd-1 fCeXHk"
  >
    
    12
  </span>
  round
</div>
`;
